SP5WWP/m17-packet/m17-packet-encode.c

```diff
@@ -300,11 +326,22 @@ int main(int argc, char* argv[])
             fill_data(full_packet, &pkt_sym_cnt, rf_bits);
 
             num_bytes-=25;
+            len++;
         }
         else
         {
+            //debug num_bytes here
+            // fprintf (stderr, " Last num: %d; ", num_bytes);
+
             memcpy(pkt_chunk, &full_packet_data[pkt_cnt*25], num_bytes);
             memset(&pkt_chunk[num_bytes], 0, 25-num_bytes); //zero-padding
+
+            //subtract 2 for the CRC? or does the CRC count as significant bytes of the last block
```

Comment:
I believe they *do* count. Need to check against the spec. Description might be vague - update might be required.

Reply:
I think I made this change very specifically because it was causing the extra garbage glyphs to be added to the end of the SMS string in SDRAngel and also in *FME when decoding SMS packet data, and with the vague description, I just came to the conclusion that the CRC was not in the significant portion of the last block, since the decoder needed to know where the breaking point of data to CRC is. Or maybe it IS significant, and this needed a -2 to get a consistent CRC checksum between here and the decoder, and also in FME to all get the same thing.

Again, changes I made months back originally, so I don't have the clearest memory regarding them.

Reply:
I looked at this again briefly, I need to revisit this portion when I can wrap my brain around it again (brain dead at the moment). I'm thinking now that the change to -2 here and the matching change in decode with the +2 may not have been necessary. I may, out of convenience, end up closing this pull request so I can work out a few kinks in the last commit and re-evaluate what stays and what goes.